apps/hash-external-services/docker-compose.yml

```diff
@@ -273,3 +276,91 @@ services:
       OPENAI_API_KEY: "${OPENAI_API_KEY}"
     ports:
       - "${HASH_AGENT_RUNNER_PORT}:80"
+
+  temporal-migrate:
+    build:
+      context: ./temporal
+      dockerfile: migrate.Dockerfile
+    depends_on:
+      postgres:
+        condition: service_healthy
+    environment:
+      DB: "postgresql"
+      DBNAME: "${HASH_TEMPORAL_PG_DEV_DATABASE}"
+      VISIBILITY_DBNAME: "${HASH_TEMPORAL_VISIBILITY_PG_DEV_DATABASE}"
+      DB_PORT: "5432"
+      # Intentionally use the POSTGRES user as it's the "superadmin" which has access to schema
+      POSTGRES_USER: "${POSTGRES_USER}"
+      POSTGRES_PWD: "${POSTGRES_PASSWORD}"
+      POSTGRES_SEEDS: "postgres" # the hostname of the postgres container
+      ENABLE_ES: "${HASH_OPENSEARCH_ENABLED}"
+      ES_SEEDS: "hash-dev-opensearch" # the hostname of the opensearch container
+      ES_VERSION: "v7"
+
+  temporal-setup:
+    build:
+      context: ./temporal
+      dockerfile: setup.Dockerfile
+    depends_on:
+      postgres:
+        condition: service_healthy
+      temporal:
+        condition: service_started
+    environment:
+      DB: "postgresql"
+      DBNAME: "${HASH_TEMPORAL_PG_DEV_DATABASE}"
+      VISIBILITY_DBNAME: "${HASH_TEMPORAL_VISIBILITY_PG_DEV_DATABASE}"
+      DB_PORT: "5432"
+      # Intentionally use the POSTGRES user as it's the "superadmin" which has access to schema
+      POSTGRES_USER: "${POSTGRES_USER}"
+      POSTGRES_PWD: "${POSTGRES_PASSWORD}"
+      POSTGRES_SEEDS: "postgres" # the hostname of the postgres container
+      ENABLE_ES: "${HASH_OPENSEARCH_ENABLED}"
+      ES_SEEDS: "hash-dev-opensearch" # the hostname of the opensearch container
+      ES_VERSION: "v7"
+      TEMPORAL_CLI_ADDRESS: temporal:7233
+      SKIP_DEFAULT_NAMESPACE_CREATION: "false" # left as a convenience as most temporal tooling expects default namespace
+
+  temporal:
```

Comment:
Since this is the development compose, we don't worry about the writable filesystem. I did try to do this, but it leads to a cascade of places I have to add tmpfs entries. For now this is not done, but for a prod deployment we should do this.